mapeditor/mapview.cpp

```diff
@@ -0,0 +1,448 @@
+#include "StdInc.h"
+#include "mapview.h"
+#include "mainwindow.h"
+#include <QGraphicsSceneMouseEvent>
+#include "mapcontroller.h"
+
+MinimapView::MinimapView(QWidget * parent):
+	QGraphicsView(parent)
+{
+	// Disable scrollbars
+	setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+	setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+}
+
+void MinimapView::dimensions()
+{
+	fitInView(0, 0, controller->map()->width, controller->map()->height, Qt::KeepAspectRatio);
+}
+
+void MinimapView::setController(MapController * ctrl)
+{
+	controller = ctrl;
+}
+
+void MinimapView::mouseMoveEvent(QMouseEvent *mouseEvent)
+{
+	this->update();
+	
+	auto * sc = static_cast<MinimapScene*>(scene());
+	if(!sc)
+		return;
+	
+	int w = sc->viewport.viewportWidth();
+	int h = sc->viewport.viewportHeight();
+	auto pos = mapToScene(mouseEvent->pos());
+	pos.setX(pos.x() - w / 2);
+	pos.setY(pos.y() - h / 2);
+	
+	QPointF point = pos * 32;
+			
+	emit cameraPositionChanged(point);
+}
+
+void MinimapView::mousePressEvent(QMouseEvent* event)
+{
+	mouseMoveEvent(event);
+}
+
+MapView::MapView(QWidget * parent):
+	QGraphicsView(parent),
+	selectionTool(MapView::SelectionTool::None)
+{
+}
+
+void MapView::cameraChanged(const QPointF & pos)
+{
+	//ui->mapView->translate(pos.x(), pos.y());
+	horizontalScrollBar()->setValue(pos.x());
+	verticalScrollBar()->setValue(pos.y());
+}
+
+
+void MapView::setController(MapController * ctrl)
+{
+	controller = ctrl;
+}
+
+void MapView::mouseMoveEvent(QMouseEvent *mouseEvent)
+{
+	this->update();
+
+	auto * sc = static_cast<MapScene*>(scene());
+	if(!sc || !controller->map())
+		return;
+
+	auto pos = mapToScene(mouseEvent->pos()); //TODO: do we need to check size?
+	int3 tile(pos.x() / 32, pos.y() / 32, sc->level);
+
+	if(tile == tilePrev) //do not redraw
+		return;
+
+	tilePrev = tile;
+
+	//main->setStatusMessage(QString("x: %1 y: %2").arg(QString::number(pos.x()), QString::number(pos.y())));
```

Comment:
Not needed? Lets remove

Reply:
I'd restore - it's the one way to know coordinates